File: day-01/star1.py


def remove_lettres(lines):
    res = []
    for line in lines:
        nl = ""
        for c in line:
            if ord(c) >= ord('0') and ord(c) <= ord('9'):
                nl += c
        res.append(nl)
    return res


def fct(lines):
    nl = remove_lettres(lines)
    sm = 0
    for l in nl:
        if len(l) == 0:
            continue
        c1 = ord(l[0]) - ord('0')
        c2 = ord(l[len(l) - 1]) - ord('0')
        sm += c1*10 + c2
    return str(sm)


f = open("input-exple", "r")
lines = f.readlines()

print("result exple: " + fct(lines))

f = open("input-star1", "r")
lines = f.readlines()

print("result star1: " + fct(lines))

File: day-14/star-1.js
const fs = require('fs');

Array.prototype.trans = function () {
    return this[0].map((_, i) => this.map(x => x[i]));
};

Array.prototype.prettyForPrint = function () {
    return this.map(x => x.join("")).join("\n");
};


if (process.argv.length < 3) {
    console.log("Usage: ", process.argv[0], process.argv[1], "<input file>")
    process.exit(1)
}

// will modify line /!\
const roll = (line) => {
    const len = line.length;
    let last_good_place = 0;
    let weight = 0;
    for (let i = 0; i < len; ++i) {
        if (line[i] == "#") {
            //weight += len - i; only rounded rock :)
            last_good_place = i + 1;
        } else if (line[i] == "O") {
            weight += len - last_good_place;
            line[i] = ".";
            line[last_good_place] = "O";
            last_good_place++;
        }
    }
    return weight;
}

const lines = fs.readFileSync(process.argv[2])
    .toString()
    .split("\n")
    .filter(l => l.length)
    .map(l => l
        .split("")
    ).trans();


console.log(lines.trans().prettyForPrint(), "\n\n")


const weights = lines.map(line => roll(line));

console.log(lines.trans().prettyForPrint())

const res = weights.reduce((s, c) => s + c, 0);
console.log("result :", res);
